install/db/dbmigrate_3.7-3.8.php

```diff
@@ -0,0 +1,322 @@
+<?php
+/***********************************************************
+ Copyright (C) 2020 Siemens AG
+ Author: Gaurav Mishra <mishra.gaurav@siemens.com>
+
+ This program is free software; you can redistribute it and/or
+ modify it under the terms of the GNU General Public License
+ version 2 as published by the Free Software Foundation.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License along
+ with this program; if not, write to the Free Software Foundation, Inc.,
+ 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ ***********************************************************/
+
+/**
+ * @file
+ * @brief Migrate DB from release 3.7.0 to 3.8.0
+ */
+
+use Fossology\Lib\Db\DbManager;
+
+/**
+ * Maximum rows to process at once
+ * @var integer MAX_ROW_SIZE
+ */
+const MAX_ROW_SIZE = 100;
+
+/**
+ * Tables which needs to be recoded
+ * @var array ENCODE_TABLES
+ */
+const ENCODE_TABLES = array(
+  "copyright",
+  "ecc",
+  "keyword"
+);
+
+/**
+ * Calculate the total number of records in concerned tables
+ * @param DbManager $dbManager
+ */
+function calculateNumberOfRecordsInCopyright($dbManager)
+{
+  $sql = "SELECT count(*) AS cnt FROM ";
+  $statement = __METHOD__ . ".getCountsFor";
+  $count = 0;
+  foreach (ENCODE_TABLES as $table) {
+    $row = $dbManager->getSingleRow($sql . $table, array(), $statement . $table);
+    $count += intval($row['cnt']);
+  }
+  return $count;
+}
+
+/**
+ * Update the recoded values back into DB
+ * @param DbManager $dbManager
+ * @param string $table Table to be updated
+ * @param array $rows   Rows with ID and recoded values
+ * @return int Count of updated rows
+ */
+function updateRecodedValues($dbManager, $table, $rows)
+{
+  if (empty($rows)) {
+    return 0;
+  }
+
+  $values = array();
+  foreach ($rows as $id => $content) {
```

Comment:
can we use Array_map here ??

Reply:
I don't think that will be necessary as this is just 2 lines of code and `array_map` requires a callback function to work on.

Also, to pass the keys, I will have to call `array_keys` to get the keys which will have it's own overhead.

Reply:
Actually inbuilt functions will be faster compared to loops. can improve speed a bit.

Reply:
Done, please check.